watertap/tools/parameter_sweep/tests/test_parameter_sweep.py

```diff
@@ -917,6 +917,110 @@ def test_parameter_sweep_optimize(self, model, tmp_path):
             _assert_dictionary_correctness(truth_dict, read_dict)
             _assert_h5_csv_agreement(csv_results_file_name, read_dict)
 
+    @pytest.mark.component
+    def test_parameter_sweep_optimize_with_added_var(self, model, tmp_path):
+        # this will run a solve function that adds a variable but only in some
+        # of the solves.
+        """THIS TEST IS DESIGNED FOR 2 Parallel workers!!!!!!"""
```

Comment:
What would be needed to make this test work with an arbitrary number of workers? This could be dealt with in a subsequent PR.

Reply:
Its a minum number - if you have 1 worker then we don't test for missing keys. I think we should really just change how we store results from each solve (e.g instead of storing all results in arrays inside a result dictionory, we just store the value for each parameter in a dict and store an array of dicts--not sure how MPI safe that is but that would solve this issue.